py/desiutil/io.py

```diff
@@ -0,0 +1,72 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+# -*- coding: utf-8 -*-
+"""
+==================
+desiutils.io
+==================
+
+Module for I/O related code
+"""
+from __future__ import (print_function, absolute_import, division,
+                        unicode_literals)
+import numpy as np
+
+try:
+    basestring
+except NameError:  # For Python 3
+    basestring = str
+
+def yamlify(obj, debug=False):
+    """ Recursively process an object so it can be serialised for yaml
+    Based on jsonify in linetools
+
+    Note: All keys in dicts are converted to str type
+
+    Parameters
+    ----------
+    obj : any object
+    debug : bool, optional
+
+    Returns
+    -------
+    obj - the same obj is yaml_friendly format
```

Comment:
What does "the same obj is yaml_friendly format" mean?  This would be better as:
```
Returns
-------
object
    An object suitable for yaml serialization.  For example :class:`numpy.ndarray` is converted to :class:`list`, :class:`numpy.int64` is converted to :class:`int`, etc.